src/web/app/pages-student/student-profile-page/upload-edit-profile-picture-modal/upload-edit-profile-picture-modal.component.ts

```diff
@@ -18,146 +13,115 @@ import { StudentProfilePictureResults } from '../../../../types/api-output';
-              private statusMessageService: StatusMessageService) { }
+              private httpRequestService: HttpRequestService) { }
 
   ngOnInit(): void {
-    if (this.pictureKey) {
-      this.getProfilePicture().subscribe((resp: any) => {
-        this.blobToBase64Image(resp);
-      });
-      this.isImageLoaded = false;
-    }
+    this.getProfilePicture().subscribe((resp: any) => {
+      this.blobToBase64Image(resp);
+    });
   }
 
   /**
-   * Populates form data with the image blob.
+   * Gets the profile picture as blob image.
    */
-  populateFormData(file: File): void {
-    this.formData = new FormData();
-    this.formData.append('studentprofilephoto', file, file.name);
+  getProfilePicture(): Observable<Blob> {
+    const profilePicEndPoint: string = '/student/profilePic';
+    return this.httpRequestService.get(profilePicEndPoint, {}, 'blob');
   }
 
   /**
-   * Handles event(s) when a file is selected from the user's file browser.
+   * Converts the blob image into a base64 string to be shown in the image cropper.
    */
-  onFileChanged(event: any): void {
-    const file: File = event.target.files[0];
-    if (file) {
-      this.fileName = file.name;
-      this.isFileSelected = true;
-      this.populateFormData(file);
-    }
-  }
+  blobToBase64Image(image: Blob): void {
+    const reader: FileReader = new FileReader();
+    reader.addEventListener('load', () => {
+      this.croppedImage = reader.result;
+    }, false);
 
-  /**
-   * Uploads the picture that has been newly uploaded/edited.
-   */
-  uploadPicture(): void {
-    const paramsMap: { [key: string]: string } = {
-      user: this.user,
-    };
-    this.httpRequestService.post('/students/profilePic', paramsMap, this.formData)
-        .subscribe((response: StudentProfilePictureResults) => {
-          this.statusMessageService.showSuccessMessage('Your profile picture has been saved successfully');
-          this.pictureKey = response.pictureKey;
-          this.profilePicLink = `${this.backendUrl}/webapi/students/profilePic?blob-key=${this.pictureKey}`;
-
-          // Gets the updated picture as blob to be filled in the image cropper
-          this.getProfilePicture().subscribe((resp: any) => {
-            this.blobToBase64Image(resp);
-            this.imageUpdated.emit(this.pictureKey);
-          });
-
-          // Reset upload section
-          this.fileName = 'No File Selected';
-          this.isFileSelected = false;
-        }, (response: ErrorMessageOutput) => {
-          this.statusMessageService.showErrorMessage(response.error.message);
-        });
+    if (image) {
+      reader.readAsDataURL(image);
+    }
   }
 
   /**
-   * Triggers the appropriate actions when the 'Save Edited Photo' button is clicked.
+   * Populates form data with the image blob.
    */
-  saveEditedPhoto(): void {
-    this.populateFormData(this.croppedImage);
-    this.uploadPicture();
+  populateFormData(file: File): void {
```

Comment:
I am not sure why you need to call `populateFormData` every time when something changes. It looks like you can just construct the new formData in `uploadPicture()`?

It is confusing (I have to say, compared to the original) that `croppedImage ` and `formData` contain "same" data.

Reply:
You have a good point here. 

However, there is a reason for this:
1. If an Image is changed in any way, I need access to the form data to send out the POST request. Populating it this way allows me to do it.
2. `croppedImage` has only one purpose. It preloads an image into the cropper if the user already has a profile picture. Other than that, I do not have any use for it. The library is built as such.

Do let me know if there is a better way to do this.